prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -387,7 +403,73 @@ def retrieve_top_datasets(
         dataset_names = [x.name for x in sorted_list]
         return dataset_names
 
-    def rerank_datasets(self, dataset_list: list[str], prompt_spec: PromptSpec):
+    def make_dataset_from_samples(
+        self,
+        inputs: list[str],
+        outputs: list[str],
+    ) -> datasets.DatasetDict:
+        """Given a list of inputs and outputs, make a dataset.
+
+        This function takes in inputs and outputs, both as list of strings,
+        and returns a DatasetDict object with a single split, "train". It has
+        two columns, "input_col" and "output_col".
+
+
+        Args:
+            inputs: A list of inputs, each input is a string.
+            outputs: A list of outputs, each output is a string.
+
+        Returns:
+            A DatasetDict object with a single split, "train". It has two
+            columns, "input_col" and "output_col".
+        """
+        updated_inputs, updated_outputs = [], []
+        if len(inputs) < 0 or len(inputs) != len(outputs):
+            logger.error("Length of inputs and outputs must be >0 and equal.")
+        else:
+            for i, o in zip(inputs, outputs):
+                if i is not None and o is not None:
+                    updated_inputs.append(i)
+                    updated_outputs.append(o)
+                else:
+                    logger.warning(f"Input or output is None: {i} {o}")
+
+        dataset_dict = {}
+        dataset_dict["train"] = datasets.Dataset.from_dict(
+            {"input_col": updated_inputs, "output_col": updated_outputs}
+        )
+        return datasets.DatasetDict(dataset_dict)
+
+    def get_rerank_with_highest_votes(self, prompt, infos_dict):
+        """Returns the dataset/config name with the highest number of votes.
+
+        Args:
+            prompt (str): The prompt used for retrieving the dataset/config name.
+            infos_dict (dict): A dictionary containing dataset/config names as keys.
+            num_votes (int, optional): The number of votes to consider. Defaults to 3.
+
+        Returns:
+            str: The dataset/config name with the highest number of votes,
+            or None if no dataset/config is found.
+        """
+        voting = []
+
+        for _ in range(self.num_votes):
+            curr_name = parse_prompt_to_fields(prompt, module_name="rerank")
+            if curr_name["name"] not in infos_dict:
+                logger.warning("LLM hallucinated dataset/config name: %s", curr_name)
```

Comment:
might need to handle the case where all `self.num_votes` times, LLM hallucinates config and appends None. Should have a logger warning for that too.

Reply:
I think the logger warning on https://github.com/neulab/prompt2model/blob/15a61eb555f96cdf86211b044371dc1ec4a9ef6a/prompt2model/dataset_retriever/description_dataset_retriever.py#L464 covers that right?

Reply:
does it? if you're appending None everytime dataset/config is hallucinated, len(voting) will not be 0 right?

Reply:
right sorry - updated the code!